Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -276,7 +328,63 @@ extension _BSONEncoder {
         return try self.box_(value) ?? Document()
     }
 
+    fileprivate func boxDate(_ date: Date) throws -> BSONValue? {
+        switch self.options.dateEncodingStrategy {
+        case .bsonDate:
+            return date
+        case .deferToDate:
+            try date.encode(to: self)
+            return self.storage.popContainer()
+        case .millisecondsSince1970:
+            return date.msSinceEpoch
+        case .secondsSince1970:
+            return date.timeIntervalSince1970
+        case .formatted(let formatter):
+            return formatter.string(from: date)
+        case .iso8601:
+            if #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+                return BSONEncoder.iso8601Formatter.string(from: date)
+            } else {
+                throw MongoError.bsonEncodeError(message: "ISO8601DateFormatter is unavailable on this platform.")
+            }
+        case .custom(let f):
+            let depth = self.storage.count
+
+            do {
+                try f(date, self)
+            } catch {
+                if self.storage.count > depth {
+                    _ = self.storage.popContainer()
+                }
+                throw error
+            }
+
+            // If they didn't encode anything, will default to empty subdocument
```

Comment:
nit: while your comment here is true based on the context this is called from, it's confusing because we return `nil`, not empty doc. 
it might be more appropriate to move this comment into the docstring for `box` where we call this, and to add a docstring to this method (and maybe also `_box`) clarifying the overall behavior - "returns the date as a BSONValue, or nil if no values were encoded by custom strategy", or something like that.

Reply:
Done 👍